libs/community/langchain_community/document_loaders/confluence.py

```diff
@@ -476,9 +485,15 @@ def paginate_request(self, retrieval_method: Callable, **kwargs: Any) -> List:
                 ),
                 before_sleep=before_sleep_log(logger, logging.WARNING),
             )(retrieval_method)
-            batch = get_pages(**kwargs, start=len(docs))
-            if not batch:
-                break
+            if self.cql:  # cursor pagination for CQL
+                batch, next_url = get_pages(**kwargs, next_url=next_url)
+                if not next_url:
+                    docs.extend(batch)
+                    break
+            else:
+                batch = get_pages(**kwargs, start=len(docs))
```

Comment:
does this need to be `batch, _ = get(pages(...))`?

Reply:
no, because for other than cql, it doesn't return anything except batch